tools/mtouch/Application.cs

```diff
@@ -596,6 +874,163 @@ public void SetDefaultFramework ()
 			}
 		}
 
+		void DetectCodeSharing ()
+		{
+			if (AppExtensions.Count == 0)
+				return;
+
+			if (!IsDeviceBuild)
+				return;
+
+			if (NoDevCodeShare) {
+				Driver.Log (2, "Native code sharing has been disabled in the main app, so no code sharing with extensions will occur.");
+				return;
+			}
+
+			if (DeploymentTarget.Major < 8) {
+				Driver.Log (2, $"Native code sharing has been disabled, because the container app's deployment target is earlier than iOS 8.0 (its {DeploymentTarget})");
+				return;
+			}
+
+			// No I18N assemblies can be included
+			if (I18n != Mono.Linker.I18nAssemblies.None) {
+				Driver.Log (2, "Native code sharing has been disabled, because the container app includes I18N assemblies ({0}).", I18n);
+				return;
+			}
+
+			List<Application> candidates = new List<Application> ();
+			foreach (var appex in AppExtensions) {
+				if (appex.IsWatchExtension)
```

Comment:
Yes, I believe very few changes will be needed to support sharing code between intents extensions and watchkit extensions.